src/routes/safe/container/index.jsx

```diff
@@ -1,199 +1,83 @@
+      ...state,
+      [`show${action}`]: true,
+    })
   }
 
-  onHide = (action: Action) => () => {
-    this.setState(() => ({ [`show${action}`]: false }))
+  const onHide = (action: Action) => () => {
+    setState({
+      ...state,
+      [`show${action}`]: false,
+    })
   }
 
-  showSendFunds = (token: Token) => {
-    this.setState({
+  const showSendFunds = (token: Token) => {
+    setState({
+      ...state,
       sendFunds: {
         isOpen: true,
         selectedToken: token,
       },
     })
   }
 
-  hideSendFunds = () => {
-    this.setState({
+  const hideSendFunds = () => {
+    setState({
+      ...state,
       sendFunds: {
         isOpen: false,
         selectedToken: undefined,
       },
     })
   }
-
-  checkForUpdates() {
-    const {
-      activeTokens,
-      checkAndUpdateSafe,
-      fetchEtherBalance,
-      fetchTokenBalances,
-      fetchTransactions,
-      safe,
-      safeUrl,
-    } = this.props
-    checkAndUpdateSafe(safeUrl)
-    fetchTokenBalances(safeUrl, activeTokens)
-    fetchEtherBalance(safe)
-    fetchTransactions(safeUrl)
-  }
-
-  render() {
-    const { sendFunds, showReceive } = this.state
-    const {
-      activateAssetsByBalance,
-      activateTokensByBalance,
-      activeTokens,
-      addressBook,
-      blacklistedTokens,
-      cancellationTransactions,
-      createTransaction,
-      currencySelected,
-      currencyValues,
-      fetchCurrencyValues,
-      fetchTokens,
-      granted,
-      network,
-      processTransaction,
-      provider,
-      safe,
-      tokens,
-      transactions,
-      updateAddressBookEntry,
-      updateSafe,
-      userAddress,
-    } = this.props
-
-    return (
-      <Page>
-        <Layout
-          activateAssetsByBalance={activateAssetsByBalance}
-          activateTokensByBalance={activateTokensByBalance}
-          activeTokens={activeTokens}
-          addressBook={addressBook}
-          blacklistedTokens={blacklistedTokens}
-          cancellationTransactions={cancellationTransactions}
-          createTransaction={createTransaction}
-          currencySelected={currencySelected}
-          currencyValues={currencyValues}
-          fetchCurrencyValues={fetchCurrencyValues}
-          fetchTokens={fetchTokens}
-          granted={granted}
-          hideSendFunds={this.hideSendFunds}
-          network={network}
-          onHide={this.onHide}
-          onShow={this.onShow}
-          processTransaction={processTransaction}
-          provider={provider}
-          safe={safe}
-          sendFunds={sendFunds}
-          showReceive={showReceive}
-          showSendFunds={this.showSendFunds}
-          tokens={tokens}
-          transactions={transactions}
-          updateAddressBookEntry={updateAddressBookEntry}
-          updateSafe={updateSafe}
-          userAddress={userAddress}
-        />
-      </Page>
-    )
-  }
+  const { sendFunds, showReceive } = state
+
+  return (
+    <Page>
+      {!safeLoaded ? null : (
```

Comment:
what do we need this variable for?

It's only working for the first time you load a safe in the app, after that it will be true forever (it doesn't change to false when you switch to a different safe)

Reply:
it's not necessary any more, was needed before, with the new refactors it's not longer needed, I've removed it